### ISSUE TITLE
Question / Bug about HMR

### ISSUE THREAD

**Issue by jhoopes** [JeffreyWay/laravel-mix#739]
- Laravel Mix Version: 0.11.0
- Node Version: 7.9.0
- NPM Version: 4.5.0
- OS: macOS Sierra

### Description:

I had a question about HMR.  It seems that when running it via Laravel Valet, I get the Access-Control-Allow-Origin conflicts between my valet .dev domain and the localhost:8080 domain.

Running it with the standard devServer options, I get this problem, but reading up on what might cause this, I found that I can add the header to the dev server by passing a headers object on the devServer config.

```js
module.exports.devServer = {
    historyApiFallback: true,
    noInfo: true,
    compress: true,
    quiet: true,
    headers: { "Access-Control-Allow-Origin": "*"},
};
```

This gets it to work, but I'm wondering if this is a bug, or is there something that I'm missing in terms of how HMR is supposed to work?  And if we need the headers object, is this something that we'd want to add to the base webpack config?

### Steps To Reproduce:
 - Base Laravel Install with Laravel Mix
 - Run `npm run hot` with some CSS and JS compiled by mix
 - Load up the web page, with the `mix()` function being used to load the `mix-manefest.json` file
 - make a change on one of the Vue files
 - The HMR module on the web page gets a push of an update, but on update, I get the Access-Control-Allow-Origin error

To add the header
 - Copy webpack.config.js to my local directory, and update the `module.exports.devServer` object as I mentioned above
 - Repeat the process of running `npm run hot` and then I get no origin errors

Thanks for your help

**Comment by jimrubenstein** [comment]
Same issue here -- seems like this should be part of the 80% target of consumers...unless there are other plans/ways to implement Hot Module Reloading.

**Comment by KuroThing** [comment]
Rejoice, #755 was closed because the PR #748 solves this issue and was merged

Also, just a heads up, if you don't want to edit / maintain your own `webpack.config.js` you can also use 
```js
mix.webpackConfig({
	devServer: {
		headers: {
			"Access-Control-Allow-Origin": "*",
		}
	}
});
```

This will also work on other webpack options and the default `webpack.mix.js` file that comes in the setup folder contains the remark: Override webpack.config.js, without editing the file directly.
